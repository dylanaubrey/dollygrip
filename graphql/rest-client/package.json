{
  "name": "@dollygrip/rest-client",
  "version": "0.0.5",
  "description": "The Dollygrip rest-client module.",
  "homepage": "https://github.com/dylanaubrey/dollygrip",
  "bugs": {
    "url": "https://github.com/dylanaubrey/dollygrip/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/dylanaubrey/dollygrip/graphql/rest-client"
  },
  "license": "MIT",
  "author": "Dylan Aubrey",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "compile": "concurrently npm:compile:*",
    "compile-watch": "concurrently npm:compile-watch:*",
    "compile-watch:main": "npm run compile:main -- --watch",
    "compile-watch:types": "npm run compile:types -- --watch",
    "compile:main": "JS_ENV=node MODULE_SYSTEM=cjs rollup -c ./rollup.config.js",
    "compile:module": "JS_ENV=node MODULE_SYSTEM=esm rollup -c ./rollup.config.js",
    "compile:types": "tsc --build",
    "test": "jest --logHeapUsage"
  },
  "dependencies": {
    "getta": "0.1.5"
  },
  "devDependencies": {
    "@cachemap/core": "3.0.2",
    "@cachemap/map": "3.0.2",
    "@dollygrip/babel-config": "0.0.1",
    "@dollygrip/jest-config": "0.0.1",
    "@dollygrip/rollup-config": "0.0.1",
    "concurrently": "5.1.0",
    "del-cli": "3.0.0",
    "jest": "26.0.1",
    "rollup": "2.47.0"
  }
}
