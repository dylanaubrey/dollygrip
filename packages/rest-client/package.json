{
  "name": "@dollygrip/rest-client",
  "version": "0.0.5",
  "description": "The Dollygrip rest-client module.",
  "keywords": [
    "api",
    "graphql",
    "schema",
    "the movie db",
    "tmdb"
  ],
  "homepage": "https://github.com/dylanaubrey/dollygrip",
  "bugs": {
    "url": "https://github.com/dylanaubrey/dollygrip/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/dylanaubrey/dollygrip"
  },
  "license": "MIT",
  "author": "Dylan Aubrey <dylanaubrey@gmail.com>",
  "main": "lib/main/index.js",
  "module": "lib/module/index.js",
  "types": "lib/types/index.d.ts",
  "scripts": {
    "compile": "concurrently npm:compile:*",
    "compile-watch": "concurrently npm:compile-watch:*",
    "compile-watch:main": "yarn run compile:main --watch",
    "compile-watch:types": "yarn run compile:types --watch",
    "compile:main": "cross-env BABEL_ENV=main babel ./src --out-dir ./lib/main --extensions \".ts\" --source-maps --config-file ../../babel.config.js",
    "compile:module": "cross-env BABEL_ENV=module babel ./src --out-dir ./lib/module --extensions \".ts\" --source-maps --config-file ../../babel.config.js",
    "compile:types": "tsc --build"
  },
  "dependencies": {
    "@cachemap/constants": "^3.0.2",
    "@cachemap/core": "^3.0.1",
    "getta": "^0.1.2",
    "isomorphic-fetch": "^2.2.1"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.0.0",
    "core-js": "> 4",
    "lodash": "^4.17.5"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@cachemap/map": "^3.0.1",
    "@cachemap/reaper": "^3.0.1"
  }
}
