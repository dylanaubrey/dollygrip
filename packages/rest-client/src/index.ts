export * from "./constants";
export { default } from "./main";
export * from "./types";
