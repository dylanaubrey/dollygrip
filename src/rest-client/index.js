import Getta from 'getta';

/**
 *
 * @type {string}
 */
const baseURL = 'https://api.themoviedb.org/3/';

/**
 *
 * @type {Object}
 */
const queryParams = { api_key: 'fds89afd098fdasjojl3jo9doijeojfd' };

/**
 *
 * @type {Getta}
 */
const getta = new Getta({ baseURL, queryParams });

// Certifications
getta.shortcut('get', 'getMovieCertifications', { path: 'certification/movie/list' });
getta.shortcut('get', 'getTVCertifications', { path: 'certification/tv/list' });

// Collection
getta.shortcut('get', 'getCollection', { path: 'collection' });
getta.shortcut('get', 'getCollectionImages', { path: 'collection/{id}/images' });

// Company
getta.shortcut('get', 'getCompany', { path: 'company' });
getta.shortcut('get', 'getCompanyMovies', { path: 'company/{id}/movies' });

// Movie
getta.shortcut('get', 'getMovie', { path: 'movie' });

// TODO

export default getta;
