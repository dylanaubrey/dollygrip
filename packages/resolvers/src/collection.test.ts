import { COLLECTION_PATH, ShortcutMethodNames } from "@dollygrip/rest-client";
import fetchMock from "fetch-mock";
import { Getta, ShortcutProperties } from "getta";
import collectionDetails from "./__tests__/data/collection/details.json";
import buildEndpoint from "./__tests__/helpers/build-endpoint";
import createRestClient from "./__tests__/helpers/create-rest-client";
import resolveCollection from "./collection";

describe("resolveCollection >", () => {
  let restClient: Getta & ShortcutProperties<ShortcutMethodNames>;

  beforeAll(() => {
    fetchMock.get(buildEndpoint(COLLECTION_PATH, { id: "10" }), collectionDetails);
    restClient = createRestClient();
  });

  afterAll(() => {
    fetchMock.restore();
  });

  describe("GIVEN a collection id >", () => {
    it("THEN the resolver should return the collection details", async () => {
      expect(await resolveCollection({}, { id: "10" }, { restClient })).toMatchSnapshot();
    });
  });
});
