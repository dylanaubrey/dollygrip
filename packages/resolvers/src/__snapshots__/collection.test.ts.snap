// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveCollection > GIVEN a collection id > THEN the resolver should return the collection details 1`] = `
Object {
  "backdrop_path": "/d8duYyyC9J5T825Hg7grmaabfxQ.jpg",
  "id": 10,
  "name": "Star Wars Collection",
  "overview": "An epic space-opera theatrical film series, which depicts the adventures of various characters \\"a long time ago in a galaxy far, far away….\\"",
  "parts": Array [
    Object {
      "adult": false,
      "backdrop_path": "/4iJfYYoQzZcONB9hNzg0J0wWyPH.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
      ],
      "id": 11,
      "original_language": "en",
      "original_title": "Star Wars",
      "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
      "popularity": 38.015,
      "poster_path": "/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg",
      "release_date": "1977-05-25",
      "title": "Star Wars",
      "video": false,
      "vote_average": 8.2,
      "vote_count": 12962,
    },
    Object {
      "adult": false,
      "backdrop_path": "/amYkOxCwHiVTFKendcIW0rSrRlU.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
      ],
      "id": 1891,
      "original_language": "en",
      "original_title": "The Empire Strikes Back",
      "overview": "The epic saga continues as Luke Skywalker, in hopes of defeating the evil Galactic Empire, learns the ways of the Jedi from aging master Yoda. But Darth Vader is more determined than ever to capture Luke. Meanwhile, rebel leader Princess Leia, cocky Han Solo, Chewbacca, and droids C-3PO and R2-D2 are thrown into various stages of capture, betrayal and despair.",
      "popularity": 16.922,
      "poster_path": "/9SKDSFbaM6LuGqG1aPWN3wYGEyD.jpg",
      "release_date": "1980-05-20",
      "title": "The Empire Strikes Back",
      "video": false,
      "vote_average": 8.4,
      "vote_count": 10949,
    },
    Object {
      "adult": false,
      "backdrop_path": "/koE7aMeR2ATivI18mCbscLsI0Nm.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
      ],
      "id": 1892,
      "original_language": "en",
      "original_title": "Return of the Jedi",
      "overview": "Luke Skywalker leads a mission to rescue his friend Han Solo from the clutches of Jabba the Hutt, while the Emperor seeks to destroy the Rebellion once and for all with a second dreaded Death Star.",
      "popularity": 14.377,
      "poster_path": "/lrJWyjOVjPhghl4KyAMtOepAxs.jpg",
      "release_date": "1983-05-25",
      "title": "Return of the Jedi",
      "video": false,
      "vote_average": 8,
      "vote_count": 9108,
    },
    Object {
      "adult": false,
      "backdrop_path": "/oFmzoKDMedABPpRBDVwr0aynSK1.jpg",
      "genre_ids": Array [
        12,
        28,
        878,
      ],
      "id": 1893,
      "original_language": "en",
      "original_title": "Star Wars: Episode I - The Phantom Menace",
      "overview": "Anakin Skywalker, a young slave strong with the Force, is discovered on Tatooine. Meanwhile, the evil Sith have returned, enacting their plot for revenge against the Jedi.",
      "popularity": 18.435,
      "poster_path": "/n8V09dDc02KsSN6Q4hC2BX6hN8X.jpg",
      "release_date": "1999-05-19",
      "title": "Star Wars: Episode I - The Phantom Menace",
      "video": false,
      "vote_average": 6.4,
      "vote_count": 9125,
    },
    Object {
      "adult": false,
      "backdrop_path": "/vovhXw10DSsRS9yoVbZqb60pXyX.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
      ],
      "id": 1894,
      "original_language": "en",
      "original_title": "Star Wars: Episode II - Attack of the Clones",
      "overview": "Following an assassination attempt on Senator Padmé Amidala, Jedi Knights Anakin Skywalker and Obi-Wan Kenobi investigate a mysterious plot that could change the galaxy forever.",
      "popularity": 16.733,
      "poster_path": "/2vcNFtrZXNwIcBgH5e2xXCmVR8t.jpg",
      "release_date": "2002-05-15",
      "title": "Star Wars: Episode II - Attack of the Clones",
      "video": false,
      "vote_average": 6.5,
      "vote_count": 8268,
    },
    Object {
      "adult": false,
      "backdrop_path": "/wUYTfFbfPiZC6Lcyt1nonr69ZmK.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
      ],
      "id": 1895,
      "original_language": "en",
      "original_title": "Star Wars: Episode III - Revenge of the Sith",
      "overview": "The evil Darth Sidious enacts his final plan for unlimited power -- and the heroic Jedi Anakin Skywalker must choose a side.",
      "popularity": 19.919,
      "poster_path": "/tgr5Pdy7ehZYBqBkN2K7Q02xgOb.jpg",
      "release_date": "2005-05-17",
      "title": "Star Wars: Episode III - Revenge of the Sith",
      "video": false,
      "vote_average": 7.3,
      "vote_count": 8541,
    },
    Object {
      "adult": false,
      "backdrop_path": "/c2Ax8Rox5g6CneChwy1gmu4UbSb.jpg",
      "genre_ids": Array [
        28,
        12,
        14,
        878,
      ],
      "id": 140607,
      "original_language": "en",
      "original_title": "Star Wars: The Force Awakens",
      "overview": "Thirty years after defeating the Galactic Empire, Han Solo and his allies face a new threat from the evil Kylo Ren and his army of Stormtroopers.",
      "popularity": 32.489,
      "poster_path": "/weUSwMdQIa3NaXVzwUoIIcAi85d.jpg",
      "release_date": "2015-12-15",
      "title": "Star Wars: The Force Awakens",
      "video": false,
      "vote_average": 7.4,
      "vote_count": 13854,
    },
    Object {
      "adult": false,
      "backdrop_path": "/jOzrELAzFxtMx2I4uDGHOotdfsS.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
      ],
      "id": 181812,
      "original_language": "en",
      "original_title": "Star Wars: The Rise of Skywalker",
      "overview": "The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power and knowledge of generations behind them, the final battle begins.",
      "popularity": 64.003,
      "poster_path": "/db32LaOibwEliAmSL2jjDF6oDdj.jpg",
      "release_date": "2019-12-18",
      "title": "Star Wars: The Rise of Skywalker",
      "video": false,
      "vote_average": 6.5,
      "vote_count": 3050,
    },
    Object {
      "adult": false,
      "backdrop_path": "/5Iw7zQTHVRBOYpA0V6z0yypOPZh.jpg",
      "genre_ids": Array [
        28,
        12,
        14,
        878,
      ],
      "id": 181808,
      "original_language": "en",
      "original_title": "Star Wars: The Last Jedi",
      "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
      "popularity": 34.936,
      "poster_path": "/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
      "release_date": "2017-12-13",
      "title": "Star Wars: The Last Jedi",
      "video": false,
      "vote_average": 7,
      "vote_count": 9917,
    },
  ],
  "poster_path": "/iTQHKziZy9pAAY4hHEDCGPaOvFC.jpg",
}
`;
